### PR TITLE
Support for filtering provider offers by golem.inf.cpu.threads and golem.inf.cpu.cores

### PR DESCRIPTION
**Pull request by filipgolem**

### CODE REVIEW

File: yajsapi/package/index.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export type RepoOpts = {
   image_hash: string;
   min_mem_gib: number;
   min_storage_gib: number;
+  min_available_logical_cpu_cores?: number;
```
Comment (azawlocki) [inline yajsapi/package/index.ts:16]: I'd change the name to `min_cpu_threads`, as discussed in https://github.com/golemfactory/yapapi/pull/508

Comment (filipgolem) [reply yajsapi/package/index.ts]: Done (I reverted the "Rename variable" commit).

File: yajsapi/package/vm.ts [hunk 2]
```diff
@@ -21,19 +23,21 @@ export async function repo({
...
   min_storage_gib = 2.0,
+  min_cpu_threads = 1,
 }: RepoOpts): Promise<Package> {
   /*
     Builds reference to a demand decorator.
 
     - *image_hash*: finds package by its contents hash.
     - *min_mem_gib*: minimal memory required to execute application code.
     - *min_storage_gib* minimal disk storage to execute tasks.
+    - *min_cpu_threads*: minimal available logical CPU cores (CPU threads).
```
Comment (azawlocki) [inline yajsapi/package/vm.ts:34]:
```suggestion
    - *min_cpu_threads*: minimal number of available logical CPU cores (CPU threads).
```

**Review by azawlocki — APPROVED** [review]
LGTM!

**Pull request merged**